### ISSUE TITLE
DISCUSS: user ability to set internal thread priorities s/b rescinded

### ISSUE THREAD

**Issue by Fish-Git** [Fish-Git/hyperion#119]
Hercules is currently the only project that I know of that allows users to define the priority assignments of its internal threads.

While there may have been a valid reason for doing so at one time in the past, I cannot for the life of me think of a good reason for this to continue.

Hercules has always been a heavily multi-threaded program with the priority of its various internal threads carefully chosen and controlled to provide the best possible overall performance with respect to inter-thread coordination of certain tasks, etc. Allowing users to change (override) this carefully orchestrated ballet seems dangerous to the extreme, tantamount to allowing the average person to override a medical prescription: _"Hmmm... This prescription says to take one pill twice a day and this other one says to take two pills every 4 hours, but I think taking four pills every four hours and one pill every 8 hours instead is probably better."_

Note that allowing the user to tweak a processes "nice" value is a completely different thing altogether, and is _not_ what's under discussion here (although a valid argument could certainly be launched for removing that ability as well).

What say ye fellow Hercules developers? Should we continue to support the various "xxxPRIO" configuration file statements? (HERCPRIO, DEVPRIO, SRVPRIO, CPUPRIO, TODPRIO)

Or would it be best to just remove that ability altogether? (since I'm doubting anyone really uses it anyway)

**Comment by Peter-J-Jansen** [comment]
I have just decided for myself to no longer use any "xxxPRIO" settings in _any_ Hercules configuration file for the following reasons: (1) I have no instructions or reliable suggestions whatsoever as to which "xxxPRIO" settings should improve the Hercules default values. (2) I do not see how it could cope with scheduling differences between the different platforms Hercules is able to run on (e.g. Windows, Linux, MacOS, etc.). (3) I hope that the respective scheduling algorithms to do a fine job by themselves (e.g. using dynamic priorities) to maximize Hercules performance. (4) It avoids syntactic or semantic differences between the various Hercules versions.

In case the "xxxPRIO" configuration statement support would be dropped, my only suggestion would be to give a message saying that the statement is deprecated and ignored, so that users are not forced to alter their Hercules configuration immediately.

In short, I agree with Fish's proposal.

**Comment by Fish-Git** [comment]
Implemented by commit 585e32be3436055d9afeea0a59672fca6a557b75.
